### PR TITLE
Fix #47

### LINKED ISSUE #47: RemappingClassAdapter remaps fields twice
RemappingClassAdapter was partially broken in this commit: e9311dc423db762b8accfc15a

It now remaps fields twice.

visitField maps the field then calls super.visitField. super.visitField maps the field again.

Here's an example field which broke in real usage, getting mapped from a -> x -> guiResponder instead of just a->x.

```
net/minecraft/client/gui/GuiTextField.x
Name: a => field_146209_f => x
net/minecraft/client/gui/GuiTextField.guiResponder
Name: x => field_175210_x => guiResponder
```

### PR DESCRIPTION
**Pull request by nallar**
When overriding in RemappingClassAdapter, instead of calling super.visitXXX, call cv.visitXXX. Calling super.visitXXX performs an extra unwanted mapping.

Also removes visitOuterClass, as the super method does exactly the same thing.

### CODE REVIEW

File: src/main/java/net/md_5/specialsource/RemappingClassAdapter.java [hunk 1]
```diff
@@ -83,8 +83,11 @@ public RemappingClassAdapter(final ClassVisitor cv, final CustomRemapper remappe
     @Override
     public MethodVisitor visitMethod(int access, String name, String desc,
             String signature, String[] exceptions) {
+        if (cv == null) {
```
Comment (md-5) [inline src/main/java/net/md_5/specialsource/RemappingClassAdapter.java]: When can these be null?
It seems to me if this was null the whole class would not do anything anyway.
Should probably just be preconditioned in constructor

**Comment by nallar** [comment]
That makes sense. I copied the null checking from ClassVisitor without
considering the context.


-- 
Ross Allan

File: src/main/java/net/md_5/specialsource/RemappingClassAdapter.java (hunk 1 above)
Comment (nallar) [reply src/main/java/net/md_5/specialsource/RemappingClassAdapter.java]: Removed null checks and added to constructor as suggested

**Comment by nallar** [comment]
Thanks :)

Snapshot builds don't seem to be on sonatype any more: https://oss.sonatype.org/content/repositories/snapshots/net/md-5/SpecialSource/

Am I looking in the wrong place?

**Comment by md-5** [comment]
Should be in a minute, github webhook needed updating to https

**Pull request closed without merging**